cpp/src/DataStorm/NodeSessionI.cpp

```diff
@@ -175,10 +194,26 @@ NodeSessionI::destroy()
             _instance->getObjectAdapter()->remove(_publicNode->ice_getIdentity());
         }
 
-        for (const auto& [_, session] : _sessions)
+        if (auto nodeSessionManager = _nodeSessionManager.lock())
         {
             // Notify sessions of the disconnection, don't need to wait for the result.
-            session->disconnectedAsync(nullptr);
+            for (const auto& [_, session] : _sessions)
```

Comment:
The session proxy stored in `_sessions` is a proxy to a forwarder object in the current relay node. This is intended to pass to a peer that can not directly contact the node hosting the session object.

When this was called from the NodeForwarder the result is that the `disconnect` request, was send after the forwarder call, even if in the code the disconnect invocation is done first. This is because it goes through the forwarder and this has to wait for the forwarder to dispatch it (collocated) and send it to the target node, which isn't really necessary as the relay can directly contact the session over the incoming connection associated with the session.